Blog post about CZI talk topic on metrics and impact (#786)
ropensci/roweb3: content/blog/2024-07-16-impact-metrics/index.en.md

---
title: "Metrics, Impact and Community Management"
author: 
    - Yanina Bellini Saibene
editor:
- Steffi LaZerte
date: '2024-07-16'
slug: impact-community-management
categories:
  - blog
tags:
  - Champions program
  - Community Manager Tools
description: "Our community manager Yani, share our experience using two frameworks with a community management lens to capture metrics at different levels and participation models to try to measure rOpenSci impact."
preface: This blog post is based on the talk [**r**eproducible **Open** **Sci**ence by and for All](/events/czi-2024/) at the CZI Open Science 2024 event. 
---

> On June 14 I was invited to present at the [CZI Open Science 2024](https://chanzuckerberg.com/science/programs-resources/open-science/) event. I was asked to participate on the Case Study Session 3: Demonstrating Impact of Open Science and explore the challenges of using traditional academic metrics for measuring project impact and emphasize alternative approaches.

Author: It can't be seen if you are not registered for the event.  If they open this info in the future, we can update it.

* **Follow up with participants:** we plan to contact participants after 12 months and 24 months since they finished the program.

### Community Participation Model Framework

The second framework we use is the _Community Participation Model Framework_ that is based on [the CSCCE Participation Model](https://www.cscce.org/resources/cpm/). 

This model describe four modes of community member participation:

* **Convey/Consume:** the community *convey* content for members to *consume* independently of one another.

* **Contribute:** members are empowered to contribute in some way to the community activities.

* **Collaborate:** members of a community collaborate with one another for the community activities, often with the community providing scaffolding for success. 

* **Co-create:** members work together within the community to co-create something that they couldn’t do before and that can involve others communities.

#### Example of the Community Participation Model Framework

To exemplify this framework, let me introduce you to [Joel Nitta](/author/joel-nitta/) from Japan and [Athanasia Mowinckel](/author/athanasia-mo-mowinckel/) from Norway.

> In [their own words](https://ropensci.org/blog/2023/07/20/teaching-targets-with-penguins/):

Reviewer: ```suggestion
In [their own words](/blog/2023/07/20/teaching-targets-with-penguins/):
```

{{< figure src = "workshop.png" alt = "Screenshoot of the blog post about the workshop, the workshop material in The Carpentries Incubator and Joel teaching" caption = "<strong>Blog post about the workshop, workshop material and Joel teaching</strong>">}}

#### Tools

The tools we use to capture our memebers stories and the impact of the community are:

* **Annual maintainers surveys**.
* **Interviews**: for example the series [The Stars of R-Universe](https://ropensci.org/blog/2023/11/06/r-universe-stars-finale/)
* [**Use cases**](https://ropensci.org/usecases/): collected in a public forum, users report on how they use rOpenSci reviewed packages and those are shared in our webpage
* **Social network analysis**.

### Social network analysis

We mapped rOpenSci collaboration network in 2022.  If we filter the network of those collaboration on writing blog post, this is how the network looks like in 2014 and this is how it looks in 2022.

{{< figure src = "workshop.png" alt = "" caption = "<strong>Blog post collaboration network 2014 and 2022</strong>">}}

> If we filter even more and we look at the collaboration on writing blog post in Spanish, this is how our network looks like before 2022, that year we introduce our Multilingual Publishing project, and this is how it looks in 2024. The network of contributor not only grows but also the connections between them.  We can see three clear clusters, the champions program participants, the stars of r-universe series and translations of important post for the peer-review process. Building the infrastructure for our website to be multilingual and to be able to do translations has positively impacted the amount of content we can have in other languages, not only by multilingual people but also by localizing our existing content.  

Reviewer: ```suggestion
If we filter the network even more and look at collaborations on writing blog posts in Spanish, we can compare how our network looked before 2022, the year we introduced our Multilingual Publishing project, to how it looks now, in 2024. The network has grown, not only in contributors, but also the connections among them. We can see three clear clusters, the Champions Program participants, the interviewees in the Stars of R-Universe series and translations of important posts for the peer-review process. Creating the multilingual infrastructure on our website and developing tools for translations has positively impacted the amount of content we can have in other languages, not only by multilingual people but also by localizing our existing content.  
```

> {{< figure src = "network-2024-es.png" alt = "The collaboration network before 2022 shows 2 nodes withour interation.  The collaboration network in 2024 shows 20 nodes with edges between them.  There are 3 clear clusters, champions program participants, the stars of r-universe series and translations of important post for the peer-review process" caption = "<strong>Blog post in spanish collaboration network mapping in two different moments in time.</strong>">}}

Reviewer: ```suggestion
{{< figure src = "network-2024-es.png" alt = "The collaboration network before 2022 shows 2 nodes without any interactions.  The collaboration network in 2024 shows 20 nodes with edges between them.  There are 3 clear clusters, Champions Program participants, interviewees from the Stars of R-Universe series and translations of important post for the peer-review process" caption = "<strong>Blog posts in Spanish collaboration network mapping in two different moments in time.</strong>">}}
```

> ## Suggestion to think about metrics

Reviewer: ```suggestion
## Suggestions for using metrics
```

> ## Suggestion to think about metrics

Reviewer: Again, I think you need some sort of intro, or linking sentences at the start of this section. A conclusion would also be good!

> 

Reviewer: ```suggestion
Use and develop metrics

```

> * Use metrics based on your goals.
> * That give you valuable feedback to improve your activities and programs.
> * That inform about your impact.
> * Use a mix of quantitative and qualitative metrics.

Reviewer: ```suggestion
* Based on your goals.
* Which give you valuable feedback to improve your activities and programs.
* Which inform about your impact.
* Are a mix of quantitative and qualitative.
```

The package implements _Karel the robot_ to teach introductory programming concepts in both Spanish and English. Marcos' students speak in Spanish so he want the package in their native language, at the same time he also wants karel to be used by other educators, so for that he provided additional documentation and lessons in English. 

> As the package is multilingual the editorial team considers that it is best to have one reviewer in English and one in Spanish.     
> <span style="float:right"> -- <em>[Mauro Lepore](author/mauro-lepore/), rOpenSci Editor</em></span>

This package become the [**first rOpenSci bilingual software peer-review**](https://github.com/ropensci/software-review/issues/620). This experience helped to improve one of the most important projects at rOpenSci.

#### Open Science Community level

The [multilingual aspect of the package](https://mpru.github.io/karel/) goes beyond the documentation and lessons.  Marcos design an architecture to have function names and messages in different languages, allowing students to learn to program also using their native language.

> > This package is pushing the limits of what R can do in term of multilingual functionality.

Reviewer: I think we need the spaces at the end of the line to make the quote look right in the preview
```suggestion
> This package is pushing the limits of what R can do in term of multilingual functionality.    
```

#### Tools

The tools we use to capture our members stories and the impact on the community via this Community Participation Model are:

* **Annual maintainers surveys**: we ask our maintainers to fill out a survey with to asset how we can support their role and their experience as member of our community. This information is used to look for new maintainers, develop new content, update our developers guide, and organize community calls.  
* **Interviews**: for example the [The Stars of R-Universe](/blog/2023/11/06/r-universe-stars-finale/) series. We interview members of our community to learn about their work, their experience with rOpenSci, and how they use our technical and social infrastructure.
* [**Use cases**](/usecases/): collected in a public forum, users report on how they use rOpenSci reviewed packages and those are shared on our webpage. They serve as examples for other users and as feedback for the package authors.
* **Social network analysis**: we use network analysis to map the connections among our members and the interactions among them. We use this information to improve our programs and activities.

### Social network analysis (SNA)

> As the goal of this talk was to explore the challenges of using traditional academic metrics for measuring project impact with an *emphasis on alternative approaches*, I want to share with you some details of [how we use social network analysis to measure the impact of our work](/blog/2024/07/11/collaboration-networks/).  We choose this particular tool, because although SNA it is well developed, we have not seen many examples of its use in the context of communities of practice in the field of open initiatives with the objective of understanding the community management interventions that are carried out.

Reviewer: Love this introduction! Very clear, and very flow from the previous section.

> As the goal of this talk was to explore the challenges of using traditional academic metrics for measuring project impact with an *emphasis on alternative approaches*, I want to share with you some details of [how we use social network analysis to measure the impact of our work](/blog/2024/07/11/collaboration-networks/).  We choose this particular tool, because although SNA it is well developed, we have not seen many examples of its use in the context of communities of practice in the field of open initiatives with the objective of understanding the community management interventions that are carried out.

Reviewer: ```suggestion
As the goal of this talk was to explore the challenges of using traditional academic metrics for measuring project impact with an *emphasis on alternative approaches*, I want to share with you some details of [how we use social network analysis to measure the impact of our work](/blog/2024/07/11/collaboration-networks/). We wanted to highlight this particular tool, because although SNA it is well developed, we have not seen many examples of its use in the context of communities of practice in the field of open initiatives with the objective of understanding the community management interventions that are carried out.
```

> In 2022, we mapped the rOpenSci collaboration network across years.  If we filter the network to collaborations on writing blog posts, we can compare the network of 2014 to 2022.

Reviewer: Do you want to mention what the colours mean, or make any other conclusions? I feel that if you're going to show the figure, you should give at least one example of what we learned by looking at it. Otherwise why show it? If you think that the second example is the most important, you could always drop this first one entirely 😁 
```suggestion
In 2022, we mapped the rOpenSci collaboration network across years.  If we filter the network to collaborations on writing blog posts, we can compare the network of 2014 to 2022. We can see that the network has grown in both contributions and collaborations.
```

{{< figure src = "network-2014-2022.png" alt = "Network map for 2014 blog post showing 14 nodes and few collaboration among most of them and network map for 2022 blog post with more than 30 nodes and a more dense iteraction links with several nodes having more than one collaboration." caption = "<strong>Blog post collaboration network 2014 and 2022</strong>">}}

> If we filter the network even more and look at collaborations on writing blog posts in Spanish, we can compare how our network looked before 2022, the year we introduced our Multilingual Publishing project, to how it looks now, in 2024. The network has grown, not only in contributors, but also the connections among them. We can see three clear clusters, the Champions Program participants, the interviewees in the Stars of R-Universe series and translations of important posts for the peer-review process. 

Reviewer: ```suggestion
If we filter the network in a different way to look at collaborations on writing blog posts in Spanish, we can compare how our network looked before 2022, the year we introduced our Multilingual Publishing project, to how it looks now, in 2024. The network has grown, not only in contributors, but also the connections among them. We can see three clear clusters, the Champions Program participants, the interviewees in the Stars of R-Universe series and translations of important posts for the peer-review process. 
```

{{< figure src = "network-2024-es.png" alt = "The collaboration network before 2022 shows 2 nodes without any interactions.  The collaboration network in 2024 shows 20 nodes with edges between them.  There are 3 clear clusters, Champions Program participants, interviewees from the Stars of R-Universe series and translations of important post for the peer-review process" caption = "<strong>Blog posts in Spanish collaboration network mapping in two different moments in time.</strong>">}}

> Creating the multilingual infrastructure on our website and developing tools for translations has positively impacted the amount of content we can have in other languages, not only by multilingual people but also by localizing our existing content. It also has increased the number of new contributors and opens new paths for contributions: writing and translate content. 

Reviewer: ```suggestion
Creating the multilingual infrastructure on our website and developing tools for translations has positively impacted the amount of content we can have in other languages, not only by multilingual people but also by localizing our existing content. It also has increased the number of new contributors and opens new paths for contributions: writing and translating content. 
```

Using the SNA tool we can see and understand the impact of our work in the community.

## Suggestions for using metrics

Use and develop metrics

> * **Based on your goals:** every metric requires effort to be recorded, analyzed and maintained over time.  It is important to select those that give us information about our objectives. For example, with the champions program we mapped the countries of our applicants to see if we were reaching the geographic regions we wanted to reach and after the first call we changed our communication strategy to reach more countries in Africa, Asia and Oceania.

Reviewer: ```suggestion
* **Based on your goals:** every metric requires effort to be recorded, analyzed and maintained over time.  It is important to select those that give us information about our objectives. For example, with the Champions Program we mapped the countries of our applicants to see if we were reaching the geographic regions we wanted to reach. In response, we changed our communication strategy in the second call to the program to reach more countries in Africa, Asia and Oceania.
```

> * **Which give you valuable feedback to improve your activities and programs:** metrics should be used to make decisions about our programs and content.  For example, using the answers to our maintainers survey we have a series of blog post calling for different type of [support](/blog/2024/03/01/qualtrics-call-help/) for several packages, [like answering questions](/blog/2024/02/29/targets-call-help/) and [help to close issues](/blog/2024/02/27/assertr-call-help/) and another one with recommendations around [communication](/blog/2024/05/17/communication-tips-oss-project/) and [marketing](/blog/2024/03/07/package-marketing/) for open source projects.

Reviewer: ```suggestion
* **Which give you valuable feedback to improve your activities and programs:** metrics should be used to make decisions about programs and content.  For example, using the answers to our maintainers survey we created a series of blog posts calling for different type of [support](/blog/2024/03/01/qualtrics-call-help/) for several packages, such as [answering questions](/blog/2024/02/29/targets-call-help/) and [helping to close issues](/blog/2024/02/27/assertr-call-help/). We also wrote a series of posts with recommendations for [communication](/blog/2024/05/17/communication-tips-oss-project/) and [marketing](/blog/2024/03/07/package-marketing/) for open source projects.
```

> * **Which inform about your impact:** if we can understand how, at what level and what kind of impact we have, we can look for ways to replicate it or make the necessary changes to get closer to our objectives.  From the data and stories we gather, we can say that our Slack and our Community Calls are an important space for rOpenSci people to meet and collaborate. It is not only a place to learn or to update our knowledge, but also to generate connections that go beyond the activities of rOpenSci. 

Reviewer: I think of these as suggestions for the audience, which is why I suggest using "you" rather than "we".
```suggestion
* **Which inform about your impact:** if you can understand how, at what level, and what kind of impact you have, you can look for ways to replicate it or make the necessary changes to get closer to your objectives. For example, from the data and stories we gather, we can say that our Slack and our Community Calls are an important space for rOpenSci people to meet and collaborate. It is not only a place to learn or to improve our knowledge, but also to generate connections that go beyond the activities of rOpenSci. 
```

## Conclusion

> The main recommendation is to use metrics that combine quantitative and qualitative tools to be able to make decisions about your programs and content.  It is important to *record the numbers but also the stories*.  There are many tools that can give us valuable information if we use them knowing that we will always have an incomplete picture, but it can be a useful one.  

Reviewer: ```suggestion
The main recommendation is to use metrics that combine quantitative and qualitative tools to be able to make decisions about your programs and content.  It is important to *record the numbers but also the stories*.  There are many tools which can give us valuable information if we use. Although we will always have an incomplete picture, it can still be a useful guide.  
```
